### DISCUSSION TITLE
some attempt on unknown LCD driver

### DISCUSSION THREAD

**Discussion by fubax** [Bodmer/TFT_eSPI#2598] [Compatible displays and setup files]
Hello Bodmer, and thank you for your great LCD library. Recently, I purchased two Nokia 230 LCDs and tried to run them with my ESP32 using your library. Fortunately, they worked well with the ILI9341, ILI9481, and ILI9488 drivers. After this successful attempt, I decided to purchase more LCDs for my project.  
As a rookie I don't know how to add image,here is it's link:
[https://ibb.co/1zzvC67](url)

However, upon receiving them, I noticed that some were manufactured by a different company. When I connected them to my ESP32, sometimes they worked and sometimes they did not. The odd part of the story is that if I try to reset the ESP32 with the onboard reset button, the image appears briefly, but then the display goes blank and turns white. As I do not know which driver IC is used in these LCDs, I am including the READ_ID code result here. I hope you can help me understand and solve the problem.
 
 **one of those LCD's which works good:**
 `reg(0x0000) 00 00	ID: ILI9320, ILI9325, ILI9335, ...
 
reg(0x0004) 21 21 40 40	Manufacturer ID

reg(0x0009) 20 20 61 00 00	Status Register

reg(0x000A) 08 08	Get Power Mode

reg(0x000C) 66 66	Get Pixel Format

reg(0x0061) 00 00	RDID1 HX8347-G

reg(0x0062) 00 00	RDID2 HX8347-G

reg(0x0063) 00 00	RDID3 HX8347-G

reg(0x0064) 00 00	RDID1 HX8347-A

reg(0x0065) 00 00	RDID2 HX8347-A

reg(0x0066) 00 00	RDID3 HX8347-A

reg(0x0067) 00 00	RDID Himax HX8347-A

reg(0x0070) 00 00	Panel Himax HX8347-A

reg(0x00A1) 00 00 00 00 00	RD_DDB SSD1963

reg(0x00B0) 00 00	RGB Interface Signal Control

reg(0x00B4) 00 00	Inversion Control

reg(0x00B6) 00 00 00 00 00	Display Control

reg(0x00B7) 00 00	Entry Mode Set

reg(0x00BF) 00 00 00 00 00 00	ILI9481, HX8357-B

reg(0x00C0) 00 00 00 00 00 00 00 00 00	Panel Control

reg(0x00C8) 00 00 00 00 00 00 00 00 00 00 00 00 00	GAMMA

reg(0x00CC) 00 00	Panel Control

reg(0x00D0) C8 C8 C8	Power Control

reg(0x00D2) C8 C8 C8 C8 C8	NVM Read

reg(0x00D3) 21 21 40 40	ILI9341, ILI9488

reg(0x00D4) C9 C9 C9 C9	Novatek ID

reg(0x00DA) 21 21	RDID1

reg(0x00DB) 40 40	RDID2

reg(0x00DC) 40 40	RDID3

reg(0x00E0) 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00	GAMMA-P

reg(0x00E1) 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00	GAMMA-N

reg(0x00EF) 00 00 00 00 00 00	ILI9327

reg(0x00F2) 00 00 00 00 00 00 00 00 00 00 00 00	Adjust Control 2

reg(0x00F6) 00 00 00 00	Interface Control `

**And the following are two of them that have problem:**
`
**************  YY Marked  LCD *************************

reg(0x0000) 00 00       ID: ILI9320, ILI9325, ILI9335, ...

reg(0x0004) 00 E3 83 E8 Manufacturer ID

reg(0x0009) 00 00 61 00 00      Status Register

reg(0x000A) 00 08       Get Power Mode

reg(0x000C) 00 06       Get Pixel Format

reg(0x0061) 00 00       RDID1 HX8347-G

reg(0x0062) 00 00       RDID2 HX8347-G

reg(0x0063) 00 00       RDID3 HX8347-G

reg(0x0064) 00 00       RDID1 HX8347-A

reg(0x0065) 00 00       RDID2 HX8347-A

reg(0x0066) 00 00       RDID3 HX8347-A

reg(0x0067) 00 00       RDID Himax HX8347-A

reg(0x0070) 00 00       Panel Himax HX8347-A

reg(0x00A1) 00 00 00 00 00      RD_DDB SSD1963

reg(0x00B0) 00 01       RGB Interface Signal Control

reg(0x00B4) 00 00       Inversion Control

reg(0x00B6) 00 00 00 00 00      Display Control

reg(0x00B7) 00 1A       Entry Mode Set

reg(0x00BF) 00 00 00 00 00 00   ILI9481, HX8357-B

reg(0x00C0) 00 31 20 00 00 00 00 00 00  Panel Control

reg(0x00C8) 00 3F 36 2C 21 24 26 23 24 25 27 28 24      GAMMA

reg(0x00CC) 00 30       Panel Control

reg(0x00D0) 00 04 0C    Power Control

reg(0x00D2) 00 00 00 00 00      NVM Read

reg(0x00D3) 00 00 00 00 ILI9341, ILI9488

reg(0x00D4) 00 00 00 00 Novatek ID

reg(0x00DA) 00 E3       RDID1

reg(0x00DB) 00 83       RDID2

reg(0x00DC) 00 E8       RDID3

reg(0x00E0) 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00     GAMMA-P

reg(0x00E1) 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00     GAMMA-N

reg(0x00EF) 00 00 00 00 00 00   ILI9327

reg(0x00F2) 00 00 00 00 00 00 00 00 00 00 00 00 Adjust Control 2

reg(0x00F6) 00 00 00 00 Interface Control
`

`
************** DG Marked LCD*****************
reg(0x0000) 00 00       ID: ILI9320, ILI9325, ILI9335, ...

reg(0x0004) 00 00 00 00 Manufacturer ID

reg(0x0009) 00 00 61 00 00      Status Register

reg(0x000A) 00 08       Get Power Mode

reg(0x000C) 00 06       Get Pixel Format

reg(0x0061) 00 00       RDID1 HX8347-G

reg(0x0062) 00 00       RDID2 HX8347-G

reg(0x0063) 00 00       RDID3 HX8347-G

reg(0x0064) 00 00       RDID1 HX8347-A

reg(0x0065) 00 00       RDID2 HX8347-A

reg(0x0066) 00 00       RDID3 HX8347-A

reg(0x0067) 00 00       RDID Himax HX8347-A

reg(0x0070) 00 00       Panel Himax HX8347-A

reg(0x00A1) 00 00 00 00 00      RD_DDB SSD1963

reg(0x00B0) 00 00       RGB Interface Signal Control

reg(0x00B4) 00 02       Inversion Control

reg(0x00B6) 00 0A 82 27 04      Display Control

reg(0x00B7) 00 06       Entry Mode Set

reg(0x00BF) 00 00 00 00 00 00   ILI9481, HX8357-B

reg(0x00C0) 00 21 00 00 00 00 00 00 00  Panel Control

reg(0x00C8) 00 00 00 00 00 00 00 00 00 00 00 00 00      GAMMA

reg(0x00CC) 00 72       Panel Control

reg(0x00D0) 00 00 00    Power Control

reg(0x00D2) 00 00 00 03 03      NVM Read

reg(0x00D3) 00 00 93 41 ILI9341, ILI9488

reg(0x00D4) 00 00 00 00 Novatek ID

reg(0x00DA) 00 00       RDID1

reg(0x00DB) 00 00       RDID2

reg(0x00DC) 00 00       RDID3

reg(0x00E0) 00 0F 16 14 0A 0D 06 43 75 33 06 0E 00 0C 09 08     GAMMA-P

reg(0x00E1) 00 08 2B 2D 04 10 04 3E 24 4E 04 0F 0E 35 38 0F     GAMMA-N

reg(0x00EF) 00 03 80 02 02 02   ILI9327

reg(0x00F2) 00 02 02 02 02 02 02 02 02 02 02 02 Adjust Control 2

reg(0x00F6) 00 01 00 00 Interface Control
`

**Comment by Bodmer** [comment]
I am not familiar with the Nokia 230 LCD display type or the register ID values for different displays.

You may get help on the Arduino forum in the Displays section as there are some knowledgable folk there.
